priv/templates/phx.gen.auth/auth_test.exs

```diff
@@ -116,6 +116,52 @@ defmodule <%= inspect auth_module %>Test do
     end
   end
 
+  describe "on_mount: mount_current_<%= schema.singular %>" do
+    test "assigns current_<%= schema.singular %> based on a valid <%= schema.singular %>_token ", %{conn: conn, <%= schema.singular %>: <%= schema.singular %>} do
+      <%= schema.singular %>_token = <%= inspect context.alias %>.generate_<%= schema.singular %>_session_token(<%= schema.singular %>)
+      session = conn |> put_session(:<%= schema.singular %>_token, <%= schema.singular %>_token) |> get_session()
+      {:cont, updated_socket} = <%= inspect schema.alias %>Auth.on_mount(:mount_current_<%= schema.singular %>, %{}, session, %Phoenix.LiveView.Socket{})
+      assert updated_socket.assigns.current_<%= schema.singular %>.id == <%= schema.singular %>.id
+    end
+
+    test "assigns nil to current_user assign if there isn't a valid <%= schema.singular %>_token ", %{conn: conn} do
+      <%= schema.singular %>_token = "invalid_token"
+      session = conn |> put_session(:<%= schema.singular %>_token, <%= schema.singular %>_token) |> get_session()
+      {:cont, updated_socket} = <%= inspect schema.alias %>Auth.on_mount(:mount_current_<%= schema.singular %>, %{}, session, %Phoenix.LiveView.Socket{})
+      assert updated_socket.assigns.current_<%= schema.singular %> == nil
+    end
+
+    test "assigns nil to current_user assign if there isn't a <%= schema.singular %>_token", %{conn: conn} do
+      session = conn |> get_session()
+      {:cont, updated_socket} = <%= inspect schema.alias %>Auth.on_mount(:mount_current_<%= schema.singular %>, %{}, session, %Phoenix.LiveView.Socket{})
+      assert updated_socket.assigns.current_<%= schema.singular %> == nil
+    end
+  end
+
+  describe "on_mount: ensure_authenticated" do
+    test "authenticates current_<%= schema.singular %> based on a valid <%= schema.singular %>_token ", %{conn: conn, <%= schema.singular %>: <%= schema.singular %>} do
+      <%= schema.singular %>_token = <%= inspect context.alias %>.generate_<%= schema.singular %>_session_token(<%= schema.singular %>)
+      session = conn |> put_session(:<%= schema.singular %>_token, <%= schema.singular %>_token) |> get_session()
+      {:cont, updated_socket} = <%= inspect schema.alias %>Auth.on_mount(:ensure_authenticated, %{}, session, %Phoenix.LiveView.Socket{})
+      assert updated_socket.assigns.current_<%= schema.singular %>.id == <%= schema.singular %>.id
+    end
+
+    test "redirects to login page if there isn't a valid <%= schema.singular %>_token ", %{conn: conn} do
```

Comment:
This test is generated:

```elixir
 test "redirects to login page if there isn't a valid user_token ", %{conn: conn} do
      user_token = "invalid_token"
      session = conn |> put_session(:user_token, user_token) |> get_session()
      socket = %Phoenix.LiveView.Socket{endpoint: TestAuthWeb.Endpoint}
      {:halt, updated_socket} = UserAuth.on_mount(:ensure_authenticated, %{}, session, socket)
      assert updated_socket.assigns.current_user == nil
    end
```
Is it a correct way to setup the socket?

Reply:
perfect!